WhitespaceAfter should support LITERAL_SYNCHRONIZED token

https://checkstyle.org/config_whitespace.html#WhitespaceAfter

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac SynchronizedSpace.java
/var/tmp $

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAfter">
            <property name="tokens" value="LITERAL_SYNCHRONIZED"/>
        </module>
    </module>
</module>

/var/tmp $ cat SynchronizedSpace.java
public class SynchronizedSpace {
    public static void main(String[] args) {
        synchronized(SynchronizedSpace.class) {
            System.out.println("started");
        }
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml SynchronizedSpace.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "LITERAL_SYNCHRONIZED" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "LITERAL_SYNCHRONIZED" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.
```

**Is your feature request related to a problem? Please describe.**
It would be nice if checkstyle could detect missing space after `synchronized` keyword.

**Describe the solution you'd like**
WhitespaceAfter could support new literal type - `LITERAL_SYNCHRONIZED`

**Additional context**
Google Java Style Guide requires whitespace separation of `(` from any reserved words
See https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace
>1. Separating any reserved word, such as `if`, `for` or `catch`, from an open parenthesis (`(`) that follows it on that line

#8122 seems related. This feature request is part of it.

#11489 and #11493 are similar requests for `try`/`finally` keywords support

--- a/src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/WhitespaceAfterTest.java
+++ b/src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/WhitespaceAfterTest.java
@@ -60,6 +60,7 @@ public class WhitespaceAfterTest extends AbstractGoogleModuleTestSupport {
             "35:16: " + getCheckMessage(clazz, message, "finally"),
             "36:38: " + getCheckMessage(clazz, message, "finally"),
             "40:16: " + getCheckMessage(clazz, message, "catch"),
+            "44:9: " + getCheckMessage(clazz, message, "synchronized"),
         };
         final Configuration checkConfig = getModuleConfig("WhitespaceAfter");
         final String filePath = getPath("InputWhitespaceAfterBad.java");

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterBad.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterBad.java
@@ -39,4 +39,9 @@ public class InputWhitespaceAfterBad {
     public void check6() {
         try {} catch(Exception e){} // warn
     }
+
+    public void check7() {
+        synchronized(this) { } // warn
+        synchronized (this) { }
+    }
 }

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java
@@ -40,4 +40,8 @@ public class InputWhitespaceAfterGood {
     public void check6() {
         try {} catch (Exception e){}
     }
+
+    public void check7() {
+        synchronized (this) { }
+    }
 }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
@@ -64,6 +64,8 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  * ELLIPSIS</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
  * LITERAL_SWITCH</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SYNCHRONIZED">
+ * LITERAL_SYNCHRONIZED</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRY">
  * LITERAL_TRY</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA">
@@ -105,6 +107,9 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  *
  *      try {} catch (Exception e){} // OK
  *      try {} catch(Exception e){} // violation ''catch' is not followed by whitespace'
+ *
+ *      synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
+ *      synchronized (this) { } // ok
  *  }
  * </pre>
  * <p>
@@ -181,6 +186,7 @@ public class WhitespaceAfterCheck
             TokenTypes.DO_WHILE,
             TokenTypes.ELLIPSIS,
             TokenTypes.LITERAL_SWITCH,
+            TokenTypes.LITERAL_SYNCHRONIZED,
             TokenTypes.LITERAL_TRY,
             TokenTypes.LAMBDA,
         };

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
@@ -11,7 +11,7 @@
  EmptyForIteratorPad&lt;/a&gt; to validate empty for iterators.
  &lt;/p&gt;</description>
          <properties>
-            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,LITERAL_FINALLY,LITERAL_CATCH,DO_WHILE,ELLIPSIS,LITERAL_SWITCH,LITERAL_TRY,LAMBDA"
+            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,LITERAL_FINALLY,LITERAL_CATCH,DO_WHILE,ELLIPSIS,LITERAL_SWITCH,LITERAL_SYNCHRONIZED,LITERAL_TRY,LAMBDA"
                       name="tokens"
                       type="java.lang.String[]"
                       validation-type="tokenSet">

--- a/src/main/resources/google_checks.xml
+++ b/src/main/resources/google_checks.xml
@@ -106,7 +106,7 @@
       <property name="tokens"
                value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE,
                     LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, LITERAL_FINALLY, DO_WHILE, ELLIPSIS,
-                    LITERAL_SWITCH, LITERAL_TRY, LITERAL_CATCH, LAMBDA"/>
+                    LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_CATCH, LAMBDA"/>
     </module>
     <module name="WhitespaceAround">
       <property name="allowEmptyConstructors" value="true"/>

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
@@ -150,6 +150,18 @@ public class WhitespaceAfterCheckTest
     }
 
     @Test
+    public void testLiteralSynchronized() throws Exception {
+        final String[] expected = {
+            "13:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "synchronized"),
+            "31:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "synchronized"),
+        };
+
+        verifyWithInlineConfigParser(
+                getPath("InputWhitespaceAfterLiteralSynchronized.java"),
+                expected);
+    }
+
+    @Test
     public void testDoWhile() throws Exception {
         final String[] expected = {
             "25:11: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "while"),

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralSynchronized.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralSynchronized.java
@@ -1,0 +1,34 @@
+/*
+WhitespaceAfter
+tokens = LITERAL_SYNCHRONIZED
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
+
+public class InputWhitespaceAfterLiteralSynchronized {
+
+    public void check1() {
+        synchronized(this) { // violation ''synchronized' is not followed by whitespace'
+        }
+        try {
+        }
+        catch (RuntimeException e){
+        }
+    }
+
+    public void check2() {
+        synchronized (this) { // ok
+        }
+        try {
+        }
+        catch (RuntimeException e){
+        }
+    }
+
+    public void check3() {
+        synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
+        synchronized (this) { }
+    }
+}

--- a/src/xdocs/config_whitespace.xml
+++ b/src/xdocs/config_whitespace.xml
@@ -2910,6 +2910,8 @@ class Bar {
                   ELLIPSIS</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
                   LITERAL_SWITCH</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SYNCHRONIZED">
+                  LITERAL_SYNCHRONIZED</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRY">
                   LITERAL_TRY</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA">
@@ -2944,6 +2946,8 @@ class Bar {
                   ELLIPSIS</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
                   LITERAL_SWITCH</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SYNCHRONIZED">
+                  LITERAL_SYNCHRONIZED</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRY">
                   LITERAL_TRY</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA">
@@ -2993,6 +2997,9 @@ public void myTest() {
 
     try {} catch (Exception e){} // OK
     try {} catch(Exception e){} // violation ''catch' is not followed by whitespace'
+
+    synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
+    synchronized (this) { } // ok
 }
           </pre>
         </div>

